### PR TITLE
build!: optimize for x86-64-v3 cpu microarchitecture (Haswell+)

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

In v1.6.2, we reverted change introducing CPU performance optimizations for x86-64 architecture.
We want these optimizations to be enabled in  final release image.

We will require CPUs to have [microarchitecture level x86-64-v3](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels)  (Haswell and newer).

We will still keep unoptimized code in debug images (the ones with `-debug` suffix) as a fallback mechanism.

## What was done?

Enabled CPU optimizations for release images in:
* Rust
* C and C++ (CFLAGS, CXXFLAGS)
* RocksDB (PORTABLE flag)

## How Has This Been Tested?

Started drive-abci inside userspace QEMU emulation 

#### Release: test that should fail on usupported x86-64 CPUs

If this test fails, it means your CPU is not supported.
This test should always succeed for all architectures different than x86-64 (I mean `arm`).

```
docker run --rm -ti dashpay/drive:1.7.0-pr.2374.1 verify ; echo $?
```

As a result:

* passed -  you should see the number **0** at the end of output
* failed - you should see number **132** at the end of output

#### Debug: test that should always pass

This test should always pass.

```
docker run --rm -ti dashpay/drive:1.7.0-pr.2374.1-debug verify ; echo $?
```

As a result, you should see the version and number **0** at the end of output

#### `x86-64-v1` (KVM model: kvm64)

* debug image (no optimizations): works
* release image: fails with `Illegal instruction`

#### `x86-64-v2` (KVM model: Nehalem)

* debug: works
* release: fails with `Illegal instruction`

#### `x86-64-v3` (KVM model: EPYC)

* debug: works
* release: works

## Breaking Changes

Dash Platform nodes must be compatible with x86-64-v3 ABI compatibility level 3(x86-64-v3). It translates into Intel Haswell CPU and newer.

Nice, but not normative ,table to help select a CPU:  https://www.qemu.org/docs/master/system/i386/cpu.html#abi-compatibility-levels-for-cpu-models

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced a new build argument in the Dockerfile for specifying the build profile.
	- Added a new target configuration for `x86_64-unknown-linux-gnu` with updated compilation flags.

- **Bug Fixes**
	- Updated Rust compilation flags for the `x86_64-unknown-linux-musl` target to enhance performance.

- **Documentation**
	- Updated package version to "1.7.0" and added new lints configuration for `tokio_unstable`.

### CODE REVIEW

**Comment by qwizzie** [comment]
According ChatGPT the microarchitecture level x86-64-v3 will exclude certain Haswell Family CPU's (maybe all of them) : 

Knipsel

Let me emphasize the following : 

> x86-64-v3 Specification: This was introduced to require certain modern CPU features like AVX2, BMI2, and more. While Haswell CPUs support many of these features, the x86-64-v3 label is used for processors starting with later microarchitectures (e.g., Skylake and beyond).

Please drop the v3 part of x86-64-v3

**Comment by lklimek** [comment]
@coderabbitai review all

File: Dockerfile [hunk 1]
```diff
@@ -114,9 +114,28 @@ RUN TOOLCHAIN_VERSION="$(grep channel rust-toolchain.toml | awk '{print $3}' | t
 ONBUILD ENV HOME=/root
 ONBUILD ENV CARGO_HOME=$HOME/.cargo
 
-# Configure Rust toolchain
+ONBUILD ARG CARGO_BUILD_PROFILE=dev
+
```
[the remark this answers is not in the document]
Comment (lklimek) [reply Dockerfile:118]: It works correctly, CARGO_BUILD_PROFILE is correctly recognized in subsequent stages.

**Comment by qwizzie** [comment]
> docker run --rm -ti dashpay/drive:1.7.0-pr.2374.1 verify ; echo $?

That works fine, thanks. It passed (0) on my VPS : 

Knipsel

**Comment by lklimek** [comment]
> > docker run --rm -ti dashpay/drive:1.7.0-pr.2374.1 verify ; echo $?
> 
> That works fine, thanks. It passes (0) on my VPS, but does not show version :

It's fine, my mistake (I changed the script but not description), will update. Thanks

**Review by shumkov — APPROVED** [review]

**Pull request merged**